File: presentation/src/test/java/com/iluwatar/presentation/DisplayedAlbumsTest.java
package com.iluwatar.presentation;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class DisplayedAlbumsTest {
  @Test
  void testAdd_true(){
    DisplayedAlbums displayedAlbums = new DisplayedAlbums();
    displayedAlbums.addAlbums("title", "artist", true, "composer");
    assertEquals("composer", displayedAlbums.getAlbums().get(0).getComposer());

  }

  @Test
  void testAdd_false(){
    DisplayedAlbums displayedAlbums = new DisplayedAlbums();
    displayedAlbums.addAlbums("title", "artist", false, "composer");
    assertEquals("", displayedAlbums.getAlbums().get(0).getComposer());
  }
}

File: presentation-model/src/test/kotlin/com/iluwatar/presentationmodel/DisplayedAlbumsTest.kt
/*
 * This project is licensed under the MIT license. Module model-view-viewmodel is using ZK framework licensed under LGPL (see lgpl-3.0.txt).
 *
 * The MIT License
 * Copyright © 2014-2022 Ilkka Seppälä
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

// ABOUTME: Unit tests for the DisplayedAlbums class.
// ABOUTME: Verifies album addition behavior for both classical and non-classical albums.
package com.iluwatar.presentationmodel

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class DisplayedAlbumsTest {
    @Test
    fun testAdd_true() {
        val displayedAlbums = DisplayedAlbums()
        displayedAlbums.addAlbums("title", "artist", true, "composer")
        assertEquals("composer", displayedAlbums.albums[0].composer)
    }

    @Test
    fun testAdd_false() {
        val displayedAlbums = DisplayedAlbums()
        displayedAlbums.addAlbums("title", "artist", false, "composer")
        assertEquals("", displayedAlbums.albums[0].composer)
    }
}
